mods/numbers/test/numbers.unit.test.ts

```diff
@@ -0,0 +1,177 @@
+  let numbers: any
+  let numberRef: any
+
+  before(() => {
+    numbers = new Numbers()
+    // TODO Create provider and app if doesn't exist
+  })
+
+  afterEach(() =>{
+    sandbox.restore();
+    sinon.restore();
+  })
+
+  it('Creating Number',async () => {
+    const numberReturn = new NumbersPB.Number()
+    numberReturn.setE164Number('test');
+    const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+      createNumber:() =>{
+        return{
+          sendMessage: ((r: any) => Promise.resolve(numberReturn)
+          )}
+      }
+    });
+    let result = await numbers.createNumber({
+        e164Number: '0000000000',
+        ingressApp: 'default'
+      })
+      
+      
+    expect(stubNumber.calledOnce).to.be.equal(true)
+    expect(numberReturn).to.be.equal(result);
+  })
+
+  it('Get a number by ref', async () =>{
+    const numberReturn = new NumbersPB.Number()
+    let action = new Numbers(); 
+    numberReturn.setE164Number('test');
+    const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+        getNumber:() =>{
+          return{
+            sendMessage: ((r: any) => Promise.resolve(numberReturn)
+            )}
+        }
+      });
+     let result = await action.getNumber('ref');
+      
+    expect(stubNumber.calledOnce).to.be.equal(true)
+    expect(numberReturn).to.be.equal(result);
+  })
+
+  it('Deleting a number', async () =>{
+    const refReturn = {} 
+    sandbox.spy();
+    const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+        deleteNumber:() =>{
+          return{
+            sendMessage: ((r: any) => Promise.resolve(refReturn)
+            )}
+        }
+      });
+     let result = await numbers.deleteNumber('ref');
+      
+    expect(stubNumber.calledOnce).to.be.equal(true)
+    expect(refReturn).to.be.equal(result);
+  })
+
+  it('Listing a number', async () =>{
+     const stubNumber = sandbox.stub(FonosService.prototype,'getService').returns({
+        listNumbers:() =>{
+          return{
+            sendMessage: ((r: any) => Promise.resolve([])
+            )}
+        }
+      });
+     let result = await numbers.listNumbers({pageSize:0,pageToken:0,view:0});      
+    expect(stubNumber.calledOnce).to.be.equal(true)
+  })
+
+  it('Should return error with aorLink and ingressApp', async () =>{
+    let request ={
+        aorLink:"test",
+        ingressApp:"test"
+    };
+    const stubDb = sandbox.stub(numbers,'getNumber').returns({})
+    const numberSpy = sinon.spy(numbers,'updateNumber')
+    try{
+        await numbers.updateNumber(request);
+    }catch(err){
+        expect(err.message).to.include('are not compatible parameters')
+
+    }
+ })
+ it('Should return error with no aorLink and ingressApp', async () =>{
+    let request ={
+    
+    };
+    const stubDb = sandbox.stub(numbers,'getNumber').returns({})
+    const numberSpy = sinon.spy(numbers,'updateNumber')
+    try{
```

Comment:
Consider using ChaiAsPromise instead of `try/catch`. For example, you could simplify with:

```
expect(numbers.updateNumber()).to.eventually.be.rejectedWith('you must provider either') 
```